packages/react/src/main-nav/MainNav.tsx

```diff
@@ -26,31 +27,22 @@ export function MainNav({
 	secondaryItems,
 	id,
 }: MainNavProps) {
-	const [isMobileMenuOpen, openMobileMenu, closeMobileMenu] =
-		useTernaryState(false);
-
 	const bestMatch = findBestMatch(
 		[...(items || []), ...(secondaryItems || [])],
 		activePath
 	);
 
 	return (
-		<Fragment>
+		<MainNavProvider>
```

Comment:
Created a provider so we don't have to pass `isMobileMenuOpen`, `openMobileMenu` and `closeMobileMenu` around.